orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/RollingRedBlackStrategy.groovy

```diff
@@ -86,72 +90,118 @@ class RollingRedBlackStrategy implements Strategy, ApplicationContextAware {
     }
 
     def findContext = baseContext + [
-        target: TargetServerGroup.Params.Target.current_asg_dynamic,
-        targetLocation: cleanupConfig.location,
+      target        : TargetServerGroup.Params.Target.current_asg_dynamic,
+      targetLocation: cleanupConfig.location,
     ]
 
     stages << newStage(
-        stage.execution,
-        determineTargetServerGroupStage.type,
-        "Determine Deployed Server Group",
-        findContext,
-        stage,
-        SyntheticStageOwner.STAGE_AFTER
+      stage.execution,
+      determineTargetServerGroupStage.type,
+      "Determine Deployed Server Group",
+      findContext,
+      stage,
+      SyntheticStageOwner.STAGE_AFTER
     )
 
     // java .forEach rather than groovy .each, since the nested .each closure sometimes omits parent context
     targetPercentages.forEach({ p ->
+      def source = getSource(targetServerGroupResolver, stageData, baseContext)
       def resizeContext = baseContext + [
-          target: TargetServerGroup.Params.Target.current_asg_dynamic,
-          action: ResizeStrategy.ResizeAction.scale_to_server_group,
-          source: getSource(targetServerGroupResolver, stageData, baseContext),
-          targetLocation: cleanupConfig.location,
-          scalePct: p,
-          pinCapacity: p < 100 // if p = 100, capacity should be unpinned
+        target        : TargetServerGroup.Params.Target.current_asg_dynamic,
+        action        : ResizeStrategy.ResizeAction.scale_to_server_group,
+        source        : source,
+        targetLocation: cleanupConfig.location,
+        scalePct      : p,
+        pinCapacity   : p < 100 // if p = 100, capacity should be unpinned
       ]
 
-      stages << newStage(
-          stage.execution,
-          resizeServerGroupStage.type,
-          "Grow to $p% Desired Size",
-          resizeContext,
-          stage,
-          SyntheticStageOwner.STAGE_AFTER
+      def resizeStage = newStage(
+        stage.execution,
+        resizeServerGroupStage.type,
+        "Grow to $p% Desired Size",
+        resizeContext,
+        stage,
+        SyntheticStageOwner.STAGE_AFTER
       )
+      stages << resizeStage
+
+      // an expression to grab newly deployed server group at runtime (ie. the server group being resized up)
+      def deployedServerGroupName = '${' + "#stage('${resizeStage.id}')['context']['asgName']" + '}'.toString()
```

Comment:
This seemed like a handy thing to be able to do ... required making `#stage` capable of working off an id vs name.